helm/qontract-reconcile/templates/template.yaml

```diff
@@ -127,8 +127,102 @@ objects:
                 key: log_group_name
           command: ["/bin/sh", "-c"]
           args:
+          # generate /etc/vector/vector.yaml
+          # and generate /fluentd/etc/fluent.conf
```

Comment:
why we generate these config files in init container instead of put it in a configmap and mount it?

Reply:
Probably one of those choices that was made years and years ago and we didn't really think about it until now

Reply:
Possibly to reference environment variables such as `POD_NAME`? Does Fluentd support the referencing of those from inside its normal rules?

Reply:
Oh that's a good point. `POD_NAME` is defined as an environment variable for the init container rather than an OpenShift Template parameter so I don't think it would work as just a `ConfigMap`: https://github.com/app-sre/qontract-reconcile/blob/52d03a93cfe8fbc4f99934ad45c498425166a821/helm/qontract-reconcile/templates/template.yaml#L118

Reply:
both fluentd and vector support env vars in config

* https://docs.fluentd.org/quickstart/faq#how-can-i-use-environment-variables-to-configure-parameters-dynamically
* https://vector.dev/docs/reference/configuration/#environment-variables

I think it's easier to read and debug for config file stored in configmap instead of shell script. If we can try it for vector, then all fluented related components can be left unchanged. But there maybe other unknown issues with that, up to you to decide.

Reply:
@hemslo There is an issue with using a `ConfigMap` because the config for Vector and Fluentd differ based on what the helm template values are. For instance, it isn't always true that logs will be sent to Slack, and so that config should be omitted. There's not a real good way to do that with a mounted ConfigMap.

Reply:
@hemslo That would make a great Jira card to follow up and reduce complexity as it is outside the scope of this card. This card's scope is already quite expansive and the final shape of the config will not have fluentd. If the spike isn't successful (ie, no suitable alternative for fluentd can be found), this will all be rolled back.

Reply:
https://issues.redhat.com/browse/APPSRE-10690